src/text.tsx

```diff
@@ -41,25 +41,51 @@ const vivinhoForScreenReaders = (
     </>
 );
 
-const makeVivinhoCharReadableForScreenReaders = (children: React.ReactNode): React.ReactNode => {
-    return React.Children.map(children, (child) => {
-        if (typeof child !== 'string') {
-            return child;
-        }
-        if (!child.includes(VIVINHO_CHAR)) {
-            return child;
-        }
-        return (
-            <>
-                {child.split(VIVINHO_CHAR).map((segment, idx) => (
-                    <React.Fragment key={idx}>
-                        {idx > 0 && vivinhoForScreenReaders}
-                        {segment}
-                    </React.Fragment>
-                ))}
-            </>
-        );
-    });
+const makeVivinhoCharReadableForScreenReaders = ({
+    children,
+    ariaLabel,
+    as,
+}: {
+    children: React.ReactNode;
+    ariaLabel?: string;
+    as?: React.ComponentType<any> | string;
+}) => {
+    // When the Text is a heading (<hx>), we set an aria-label replacing the vivinho char with "Vivo" for
+    // screen readers and hide the original text.
+    // If we used the generic solution, Safari/iOS VoiceOver would read "Vivo" as a separated heading
```

Comment:
this is the same Safari/VoiceOver bug we faced with links. Screen reader reads every element as a separated one, giving the impression there are multiple headings